### PR TITLE
optimize rerank with backend ref

### PR DESCRIPTION
**Pull request by a32543254**
## Description

Optimize the rerank with backend ref module
image
<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/dongb/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/dongb/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">
<style>

</style>
</head>

<body link="#467886" vlink="#96607D">

model type | input shape | Mosec (s) | NS (s) | ratio
-- | -- | -- | -- | --
bge-reranker-large | [2, 14] | 0.0537 | 0.0359 | 1.5x
bge-reranker-large | [24, 512] | 2.610966 | 1.184117 | 2.2x

</body>

</html>

## Issues

List the issue or RFC link this PR is working on. If there is no such link, please mark it as `n/a`.

## Type of change

List the type of change like below. Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would break existing design and interface)
- [ ] Others (enhancement, documentation, validation, etc.)

## Dependencies

List the newly introduced 3rd party dependency if exists.

## Tests

Describe the tests that you ran to verify your changes.

### CODE REVIEW

**Comment by lvliang-intel** [comment]
@a32543254,

Please fix pre-commit issue and also add a test case.
image

**Comment by lvliang-intel** [comment]
@a32543254，
Please fix the Docker file issue.
image

**Comment by a32543254** [comment]
> @a32543254， Please fix the Docker file issue. image

Thanks, I'm co-working with Suyue on this ut test now, will be fixed soon.

File: comps/reranks/neural-speed/reranking_neuralspeed_svc.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    if input.retrieved_docs:
+        docs = [doc.text for doc in input.retrieved_docs]
+        url = mosec_reranking_endpoint + "/inference"
+        data = {"query": input.initial_query, "docs": docs}
+        resp = requests.post(url, data=msgspec.msgpack.encode(data))
+        response = msgspec.msgpack.decode(resp.content)["scores"]
+
+        best_response_index = max(enumerate(response), key=lambda score: score[1])[0]
+        doc = input.retrieved_docs[best_response_index]
+        if doc.text and len(re.findall("[\u4E00-\u9FFF]", doc.text)) / len(doc.text) >= 0.3:
```
Comment (lvliang-intel) [inline comps/reranks/neural-speed/reranking_neuralspeed_svc.py:50]: Now the rerank microservice will not handle the chat template, please align with https://github.com/opea-project/GenAIComps/blob/main/comps/reranks/tei/reranking_tei.py

Comment (a32543254) [reply comps/reranks/neural-speed/reranking_neuralspeed_svc.py:50]: I‘m actually aligned with https://github.com/opea-project/GenAIComps/tree/main/comps/reranks/langchain-mosec. 
So I better align with TEI right ？

Comment (lvliang-intel) [reply comps/reranks/neural-speed/reranking_neuralspeed_svc.py:50]: Only the chat template part needs to align with TEI, now the RAG chat template is applied in LLM microservice, not in reranking microservice.

Comment (a32543254) [reply comps/reranks/neural-speed/reranking_neuralspeed_svc.py]: fixed

**Review by lvliang-intel — APPROVED** [review]

**Review by Spycsh — APPROVED** [review]

**Pull request merged**